src/Npgsql/NpgsqlException.cs

```diff
@@ -50,6 +51,26 @@ public NpgsqlException(string message)
         public virtual bool IsTransient =>
             InnerException is IOException || InnerException is SocketException;
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var data = Data;
```

Comment:
This is intentional because each call to `this.Data` property getter causes an expensive LINQ / reflection loop to run.